specification/changeanalysis/resource-manager/Microsoft.ChangeAnalysis/stable/2021-04-01/examples/ResourceChangesList.json

```diff
@@ -0,0 +1,68 @@
+          {
+            "id": "/subscriptions/4d962866-1e3f-47f2-bd18-450c08f914c1/resourceGroups/MyResourceGroup/providers/Microsoft.Web/sites/mysite/providers/Microsoft.ChangeAnalysis/resourceChanges/ARG_23fa00fd-dda0-4268-b482-2076825cf165_970d8c6d-6b78-4270-92ef-88d5aa2b5f0b_132316363294700000_132316498613900000",
+            "name": "ARG_23fa00fd-dda0-4268-b482-2076825cf165_970d8c6d-6b78-4270-92ef-88d5aa2b5f0b_132316363294700000_132316498613900000",
+            "type": "Microsoft.ChangeAnalysis/resourceChanges",
+            "properties": {
+              "resourceId": "/subscriptions/4d962866-1e3f-47f2-bd18-450c08f914c1/resourceGroups/MyResourceGroup/providers/Microsoft.Web/sites/mysite",
+              "timeStamp": "2021-04-26T02:17:41.39Z",
+              "initiatedByList": [
+                "ellen@contoso.com"
+              ],
+              "changeType": "Update",
+              "propertyChanges": [
+                {
+                  "changeType": "Update",
+                  "changeCategory": "User",
+                  "jsonPath": "value[1].properties.thumbprint",
+                  "displayName": "publicCertificates[\"AppCert\"].properties.thumbprint",
+                  "level": "Important",
+                  "description": "The thumbprint of the certificate",
+                  "oldValue": "21D0482F-E91E-4C14-8078-65BFDCDBCA64",
+                  "newValue": "3F2DF554-B063-4383-8BD3-4970BCF20A7E",
+                  "isDataMasked": false
+                }
+              ]
+            }
+          },
+          {
+            "id": "/subscriptions/4d962866-1e3f-47f2-bd18-450c08f914c1/resourceGroups/MyResourceGroup/providers/Microsoft.Web/sites/mysite/extensions/AppStateTracker/providers/Microsoft.ChangeAnalysis/resourceChanges/AST_b2ecd7ec-00dd-4d44-bd19-fefc88790c05_da36e22e-f541-44bd-8d89-8f10a27af4ad_132592897881782408_132592957873393845",
+            "name": "AST_b2ecd7ec-00dd-4d44-bd19-fefc88790c05_da36e22e-f541-44bd-8d89-8f10a27af4ad_132592897881782408_132592957873393845",
+            "type": "Microsoft.ChangeAnalysis/changes",
+            "properties": {
+              "resourceId": "/subscriptions/4d962866-1e3f-47f2-bd18-450c08f914c1/resourceGroups/MyResourceGroup/providers/Microsoft.Web/sites/mysite",
+              "timeStamp": "2021-04-26T02:17:41.39Z",
+              "initiatedByList": [
+                "ellen@contoso.com"
+              ],
+              "changeType": "Update",
+              "propertyChanges": [
+                {
+                  "changeType": "Update",
+                  "changeCategory": "User",
+                  "jsonPath": "environment.environmentVariables.APPSETTING_DB_CONNSTR",
+                  "displayName": "APPSETTING_DB_CONNSTR",
+                  "level": "Important",
+                  "description": "Application setting",
+                  "oldValue": "<old database connection string>",
+                  "newValue": "<new database connection string>",
```

Comment:
GET should not return a secret (e.g. connection string)

Reply:
@j5lim Is this a hard requirement that GET API should never return secret? Like ARM established the semantics that GET API means there is no secret data? 
We set this API as Unauthorized in ARM manifest, and we do the RBAC check ourselves. As the description of the API said, the customer data **will** be masked if user doesn't have permission, but if they have permission, we will show them the value.

We definitely want to show the value to the customer if they have permission, if it's a hard requirement not to use GET, what do you recommend?

Reply:
Switched to POST method.

Reply:
Yes, it is a hard requirement. https://armwiki.azurewebsites.net/rp_onboarding/process/property_design_best_practices.html?q=secret#secrets

POST should be used to return secrets.

Reply:
@j5lim Thank for the link. I also added the x-ms-secret tag.

Only the resource level query (POST /resourceChanges) may contain secret value, subscription and resource group level query (GET /changes) guarantee the secret values are masked. While in the swagger, they share the same model.
Let me know if you think we should create a different model (e.g., PropertyChangeHasSecret and PropertyChangeNoSecret) just to differentiate the x-ms-secret tag.

Reply:
I think GET response won't allow x-ms-secret set to true, so you will need to have a different model.

Reply:
@j5lim The Swagger Model validation reports error even for the POST method.
Is there anything I missed or maybe it's a bug of the validation tool? It's asking me to remove those properties, while we do want to return it.
```
SECRET_PROPERTY
Output Message: Secret property {0} cannot be sent in the response.

Description: The secret is not allowed to return in response when it's annotated with x-ms-secret:true.

How to fix the violation: Remove this secret value from the response.
```

Or maybe I misunderstood this tag? i.e., it's meant for the properties that RP never want to return regardless the GET/POST methods? Any in our case since we do want to return them, we shouldn't add this tag? I'll remove this tag for now.

Reply:
I talked to Ray Chen, we can ignore the validation error on POST method. To be more specific, I can ping him to get it approved. So I'll add the x-ms-secret tag back and create different models.